### PR TITLE
Add standalone Modal widget

### PR DESCRIPTION
**Pull request by ncordin**
# Description

For legal purposes, merchants needs to be able to open the modal without showing the payment plans on their page. This PR allow to open the Modal widget alone.

# Changes
- Add `NPM_AUTH_TOKEN` to the CI so it will be able to install `@alma/react-components`
- Update the example to show how to open the modal only
- Add a `ModalContainer` component to provide data to the `EligibilityModal` component, so it can works without `PaymentPlan`
- Copy / paste the `<LoadingIndicator />` component. Import css from `@alma/react-components` did not work with `microbundle` :(

# How to tests

- `npm start` to make a build
- run an HTTP server on the repo directory and access the example.html file

# Links

https://app.clickup.com/t/20427503/AL-328

# Screens

image

image

### CODE REVIEW

File: examples/basic.html [hunk 1]
```diff
@@ -124,7 +143,33 @@ <h3>450 €</h3>
...
+            },
+            {
+              installmentsCount: 3,
+              minAmount: 5000,
+              maxAmount: 90000,
+            },
+          ],
+        })
+
+        openReference = open
```
Comment (Taskim) [inline examples/basic.html]: Maybe we should do something like:

```
function renderModal() {
/* */
return { open, close }
}

const { open, close } = renderModal();
```

File: src/Widgets/EligibilityModal/ModalContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+type Props = {
+  purchaseAmount: number
+  apiData: ApiConfig
+  configPlans?: ConfigPlan[]
+  onClose: () => void
+}
+
+/**
+ * This component allow to display only the modal, without PaymentPlans.
```
Comment (Taskim) [inline src/Widgets/EligibilityModal/ModalContainer.tsx]:
```suggestion
 * This component allows to display only the modal, without PaymentPlans.
```

File: src/Widgets/EligibilityModal/ModalContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ */
+const ModalContainer: React.FC<Props> = ({ purchaseAmount, apiData, configPlans, onClose }) => {
+  const [eligibilityPlans, status] = useFetchEligibility(purchaseAmount, apiData, configPlans)
+  const eligiblePlans = eligibilityPlans.filter((plan) => plan.eligible)
```
Comment (Taskim) [inline src/Widgets/EligibilityModal/ModalContainer.tsx]: we should move this line to the the EligibilityModal ?

File: src/Widgets/EligibilityModal/modal.test.tsx [hunk 4]
```diff
@@ -108,4 +108,21 @@ describe('Modal', () => {
       expect(element).toHaveTextContent('150,00 €')
     })
   })
+  describe('empty plans', () => {
```
Comment (Taskim) [inline src/Widgets/EligibilityModal/modal.test.tsx:114]: We should add a test for the ModalContainer

File: src/Widgets/PaymentPlans/PaymentPlans.test.tsx [hunk 5]
```diff
@@ -2,8 +2,8 @@ import { act, fireEvent, screen, waitFor, within } from '@testing-library/react'
 import { ApiMode } from 'consts'
 import React from 'react'
 import { mockButtonPlans } from 'test/fixtures'
-import render from '../test'
-import PaymentPlanWidget from './PaymentPlan'
+import PaymentPlanWidget from '.'
+import render from '../../test'
```
Comment (Taskim) [inline src/Widgets/PaymentPlans/PaymentPlans.test.tsx]:
```suggestion
import render from 'test'
```

File: .npmrc.sample [hunk 6]
```diff
@@ -0,0 +1,7 @@
+@alma:registry=https://npm.pkg.github.com/
```
Comment (Taskim) [inline .npmrc.sample]: to be removed

File: .gitignore [hunk 7]
```diff
@@ -10,3 +10,5 @@ compiled
 .rpt2_cache
 docs
 stats.html
+.npmrc
```
Comment (Taskim) [inline .gitignore]: to be removed

File: .github/workflows/ci-cd.yaml [hunk 8]
```diff
@@ -18,6 +17,8 @@ jobs:
         uses: actions/setup-node@v1
         with:
           node-version: 16
+      - name: Prepare npm token
```
Comment (Taskim) [inline .github/workflows/ci-cd.yaml]: to be removed

File: documentation.md [hunk 9]
```diff
@@ -15,10 +15,18 @@ Your merchant id as it is found in the dashboard
 
 ## `add(widget: Alma.Widgets.PaymentPlans, options: PaymentPlansOptions )`
 
+## `add(widget: Alma.Widgets.Modal, options: ModalOptions )`
+
```
Comment (Taskim) [inline documentation.md]: update doc with examples for both case

File: src/Widgets/PaymentPlans/PaymentPlans.test.tsx (hunk 5 above)
Comment (ncordin) [reply src/Widgets/PaymentPlans/PaymentPlans.test.tsx]: done

File: examples/basic.html (hunk 1 above)
Comment (ncordin) [reply examples/basic.html]: yes 👍

File: .npmrc.sample (hunk 6 above)
Comment (ncordin) [reply .npmrc.sample]: done

File: .github/workflows/ci-cd.yaml (hunk 8 above)
Comment (ncordin) [reply .github/workflows/ci-cd.yaml]: done

File: .gitignore (hunk 7 above)
Comment (ncordin) [reply .gitignore]: done

File: src/Widgets/EligibilityModal/ModalContainer.tsx (hunk 3 above)
Comment (ncordin) [reply src/Widgets/EligibilityModal/ModalContainer.tsx]: done 👍

File: src/Widgets/EligibilityModal/modal.test.tsx (hunk 4 above)
Comment (ncordin) [reply src/Widgets/EligibilityModal/modal.test.tsx:114]: done 👍

File: documentation.md [hunk 10]
```diff
@@ -13,54 +13,95 @@ Your merchant id as it is found in the dashboard
...
-## Plan: `Plan`
+## Add Modal
+
+`add(widget: Alma.Widgets.Modal, options: ModalOptions )`
+
+```
+const {open, close} = widgets.add(Alma.Widgets.Modal, {
+    container: '#alma-widget-modal',
+    purchaseAmount: 20000
+}
```
Comment (Taskim) [inline documentation.md]:
```suggestion
})
```

File: documentation.md [hunk 11]
```diff
@@ -13,54 +13,95 @@ Your merchant id as it is found in the dashboard
...
 
-### hideIfNotEligible: `boolean` [optional, default: false]
+- hideIfNotEligible: `boolean` [optional, default: false]
 
 Totally hide the widget if set to true and no plan matches the purchase amount.
 
-## Plan: `Plan`
+## Add Modal
+
+`add(widget: Alma.Widgets.Modal, options: ModalOptions )`
```
Comment (Taskim) [inline documentation.md]:
```suggestion
`add(widget: Alma.Widgets.Modal, options: ModalOptions)`
```

File: documentation.md [hunk 12]
```diff
@@ -13,54 +13,95 @@ Your merchant id as it is found in the dashboard
...
+## Add PaymentPlans
 
-### widget: `Alma.Widgets.PaymentPlans` [required]
+`add(widget: Alma.Widgets.PaymentPlans, options: PaymentPlansOptions )`
+
+```
+widgets.add(Alma.Widgets.PaymentPlans, {
+    container: '#alma-widget-payment-plans',
+    purchaseAmount: 20000
+}
```
Comment (Taskim) [inline documentation.md]:
```suggestion
})
```

File: documentation.md [hunk 13]
```diff
@@ -13,54 +13,95 @@ Your merchant id as it is found in the dashboard
 - `Alma.ApiMode.TEST` Used to test the widget
 - `Alma.ApiMode.LIVE` Used in production mode. data will match what is provided in your dashboard
 
-## `add(widget: Alma.Widgets.PaymentPlans, options: PaymentPlansOptions )`
+## Add PaymentPlans
 
-### widget: `Alma.Widgets.PaymentPlans` [required]
+`add(widget: Alma.Widgets.PaymentPlans, options: PaymentPlansOptions )`
```
Comment (Taskim) [inline documentation.md]:
```suggestion
`add(widget: Alma.Widgets.PaymentPlans, options: PaymentPlansOptions)`
```

**Review by Taskim — APPROVED** [review]

**Pull request merged**